### ISSUE TITLE
Enabling kubeStateMetricsCore disables other clusterAgent checks

### ISSUE THREAD

**Issue by LiptonB** [DataDog/datadog-operator#307]
**Describe what happened:**
I am trying to set the clusterAgent up to collect both kubernetes events and kubernetes state metrics. I enabled the `kubeStateMetricsCore` and `collectEvents` configs in the DatadogAgent resource (see config below). However, with this config, events were not being collected. On the cluster agent that is elected leader, I ran `agent status` and saw that there was no `kubernetes_apiserver` check:
```
=========
Collector
=========

  Running Checks
  ==============
    No checks have run yet
```

Looking into the `/etc/datadog-agent/conf.d/` folder in the clusterAgent pods, I see that the only file is `kubernetes_state_core.yaml`. It seems like the `ksm-core-config` volume is hiding the files that are normally in that directory in the clusterAgent image:
```
        volumeMounts:
        - mountPath: /etc/datadog-agent/install_info
          name: installinfo
          readOnly: true
          subPath: install_info
        - mountPath: /conf.d
          name: confd
          readOnly: true
        - mountPath: /etc/datadog-agent/conf.d
          name: ksm-core-config
          readOnly: true
```

As a workaround, I was able to trigger [this branch of the code](https://github.com/DataDog/datadog-operator/blob/0b43e7e2f5d63c38d92bd5a003955c4edbc24366/controllers/datadogagent/clusteragent.go#L407-L411) by specifying an explicit config under `kubeStateMetricsCore`, which makes the volume override only the `kubernetes_state_core.yaml` file, not the whole `conf.d` directory.

**Describe what you expected:**
To be able to configure `kubeStateMetricsCore` and `collectEvents` simultaneously without specifying a kubeStateMetricsCore config, and have the agent collect both types of data.

**Steps to reproduce the issue:**
My DatadogAgent configuration looks like this:
```
apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  credentials:
    apiSecret:
      keyName: apiKey
      secretName: datadog-agent-secrets
  agent:
<...snip irrelevant config...>
  clusterAgent:
    image:
      name: "gcr.io/datadoghq/cluster-agent:latest"
    replicas: 3
    config:
      clusterChecksEnabled: true
      collectEvents: true
  features:
    kubeStateMetricsCore:
      enabled: true
```

When I add the following under the kubeStateMetricsCore object, then I get both events and state metrics:
```
      conf:
        configData: |
          cluster_check: true
          init_config:
          instances:
            - collectors:
                - pods
                - replicationcontrollers
                - statefulsets
                - nodes
                - cronjobs
                - jobs
                - replicasets
                - deployments
                - configmaps
                - services
                - endpoints
                - daemonsets
                - horizontalpodautoscalers
                - limitranges
                - resourcequotas
                - secrets
                - namespaces
                - persistentvolumeclaims
                - persistentvolumes
              telemetry: true
```

**Additional environment details (Operating System, Cloud provider, etc):**
Running on GKE, kubernetes version 1.17.17

**Comment by patjones** [comment]
We're seeing this as well/

**Comment by ahmed-mez** [comment]
Hi @LiptonB and @patjones 

Thanks for reporting this! We're very pleased about your interest in using Kube State Metrics Core!

We were aware that the Datadog Operator support for Kube State Metrics Core is not complete and [not yet documented](https://docs.datadoghq.com/integrations/kubernetes_state_core) either (as of today, July 9 2021).

With [this PR](https://github.com/DataDog/datadog-operator/pull/330), the Datadog Operator will offer full support for Kube State Metrics Core and the public documentation page will be updated accordingly once the next Operator version is released. Stay tuned!

Thank you!

**Comment by clamoriniere** [comment]
Hello, 

We have identified the problem, and it has been fixed in #335

We will start soon a new release.

**Comment by clamoriniere** [comment]
Hello @LiptonB and @patjones,

the Operator v0.7.0 has been release with the fix for this issue.
Don't hesitate to reopen the issue, it you still have the issue.

Thanks
Cedric
